src/System.Buffers/src/System.Buffers.csproj

```diff
@@ -13,8 +13,7 @@
     <IsPackable>true</IsPackable>
     <VersionPrefix>4.5.1</VersionPrefix>
     <VersionPrefix Condition="'$(IsPackable)' == 'true'">4.6.0</VersionPrefix>
-    <AssemblyVersion Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">4.0.3.0</AssemblyVersion>
-    <AssemblyVersion Condition="'$(IsPackable)' == 'true' and $([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">4.0.4.0</AssemblyVersion>
+    <AssemblyVersion Condition="'$(IsPackable)' == 'true'">4.0.4.0</AssemblyVersion>
```

Comment:
For this one we should keep it pinned, because `netcoreapp2.1` where the package expects the framework to provide the implementation, is 
```
// C:\Users\ericstj\.nuget\packages\microsoft.netcore.app\2.1.0\ref\netcoreapp2.1\System.Buffers.dll
// System.Buffers, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
```

I see we're pinning to `4.0.3.0` but the package's reference assembly was previously `4.0.2.0`.  Did we discuss this before?
```
// C:\Users\ericstj\.nuget\packages\system.buffers\4.5.1\ref\netstandard2.0\System.Buffers.dll
// System.Buffers, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
```

Reply:
Okay, so I need to bring back lines 16 and 17, but also correct the numbers and make them 4.0.2.0 and 4.0.3.0, respectively.

> Did we discuss this before?

I don't think we've talked about this property specifically. But maybe it feels like a deja vu because we have done similar adjustments in other assemblies:

https://github.com/dotnet/maintenance-packages/pull/146/files
https://github.com/dotnet/maintenance-packages/pull/102/files

I think we never noticed the wrong assembly version because this was the very first assembly that was migrated when the repo was created: https://github.com/dotnet/maintenance-packages/pull/3

Reply:
That's a runtime assembly, we need to look at the reference assembly.

Reply:
Ok from our meeting I understand this: we first check if there's an applicable ref and use that, if there's no applicable ref, then it checks if there's an applicable lib.
So the unconditioned property needs to match the ref assemblyversion.
The conditioned property (non-packable) needs to match the lib assemblyversion for .NET Framework.
The conditioned property (packable) needs to bump the previous one.